test: CharacterLayer ポートレートモード(#209)のテストが main で失敗（sprite.y が 800 にクランプ）

## 現象

`frontend/src/game/CharacterLayer.test.ts > CharacterLayer portrait mode (Issue #209) > screenHeight=800（9:16）で sprite.y が 800 * (380 / 450) ≒ 676 になる` が失敗する。

```
AssertionError: expected 800 to be close to 675.5555555555555
```

期待値 ≒675.5 に対し実測 800（= screenHeight そのまま）。ポートレートモード（9:16）の立ち絵 sprite.y が `screenHeight * (380 / ASPECT_RATIOS['16:9'].height)` にならず、画面下端にクランプされている疑い。

## 状態

- main（クリーン）で再現。**既存の壊れ**であり、特定の作業ブランチに起因しない（#259 の DDA リファクタとは無関係。切り分け済み）
- フルスイートは他 734 件すべて緑。本件のみ赤

## 次の調査

- `CharacterLayer` のポートレート時 y 算出（`ASPECT_RATIOS` 参照箇所）と、テストの期待式 `800 * (380 / 450)` のどちらが正かを確認する
- Issue #209 の実装後にアスペクト比定義（`ASPECT_RATIOS['16:9'].height`）か y クランプの仕様が変わってテストが取り残された可能性

--- a/frontend/src/game/CharacterLayer.test.ts
+++ b/frontend/src/game/CharacterLayer.test.ts
@@ -1,6 +1,5 @@
 import { describe, it, expect } from 'vitest'
-import { CharacterLayer, normalizePosition } from './CharacterLayer'
-import { ASPECT_RATIOS } from './constants'
+import { CHARACTER_Y_RATIO, CharacterLayer, normalizePosition } from './CharacterLayer'
 
 interface FadeAnimationLike {
   fromAlpha: number
@@ -118,12 +117,14 @@ describe('normalizePosition', () => {
 })
 
 describe('CharacterLayer portrait mode (Issue #209)', () => {
-  it('screenHeight=800（9:16）で sprite.y が 800 * (380 / 450) ≒ 676 になる', () => {
+  it('足元 Y は screenHeight * CHARACTER_Y_RATIO になる（縦長モードでも比率追従）', () => {
+    // CHARACTER_Y_RATIO は #210 後に 380/450 → 1.0 へ変更（足元を画面下端に下げる意図）。
+    // 期待値は定数を直に参照して陳腐化を防ぐ（旧 `800*(380/450)≒676` 直書きの教訓・#262）。
     const layer = new CharacterLayer(450, 800)
     layer.show('hero', 'normal', '中央', '/assets', { instant: true })
     const state = asInternals(layer).characters.get('hero')
     expect(state).toBeDefined()
-    expect(state!.sprite.y).toBeCloseTo(800 * (380 / ASPECT_RATIOS['16:9'].height), 5)
+    expect(state!.sprite.y).toBeCloseTo(800 * CHARACTER_Y_RATIO, 5)
   })
 })
 

--- a/frontend/src/game/CharacterLayer.ts
+++ b/frontend/src/game/CharacterLayer.ts
@@ -69,12 +69,13 @@ export function normalizePosition(position: string): string {
   return POSITION_ALIASES_JA[position] ?? POSITION_ALIASES_EN[position] ?? position
 }
 
-/** 足元 Y 座標の比率。
+/** 足元 Y 座標の比率（`characterY = screenHeight * CHARACTER_Y_RATIO`）。
  *  以前は 380/450 ≒ 0.844 (DialogBox の上端あたり) だったが、
  *  枠なし・教育動画モードでは立ち絵の下端を画面下端まで下げたほうが座りが良い。
  *  影響範囲: 全 game の立ち絵が画面下端まで下がる。既存 game が枠ありで足元位置を
- *  そのままにしたい場合は別途オプション化が要る。 */
-const CHARACTER_Y_RATIO = 1.0
+ *  そのままにしたい場合は別途オプション化が要る。
+ *  テストが期待値を直書きして陳腐化するのを防ぐため export する（#262）。 */
+export const CHARACTER_Y_RATIO = 1.0
 
 interface CharacterState {
   sprite: Sprite

